sonar-batch/src/main/java/org/sonar/batch/report/ReportPublisher.java

```diff
@@ -145,15 +149,22 @@ String upload(File report) {
     LOG.debug("Upload report");
     long startTime = System.currentTimeMillis();
     ProjectDefinition projectDefinition = projectReactor.getRoot();
-    SubmitWsRequest submitRequest = new SubmitWsRequest();
-    submitRequest.setProjectKey(projectDefinition.getKey());
-    submitRequest.setProjectName(projectDefinition.getName());
-    submitRequest.setProjectBranch(projectDefinition.getBranch());
-    submitRequest.setReport(report);
-    WsCe.SubmitResponse submitResponse = wsClient.computeEngine().submit(submitRequest);
-    long stopTime = System.currentTimeMillis();
-    LOG.info("Analysis report uploaded in " + (stopTime - startTime) + "ms");
-    return submitResponse.getTaskId();
+    PostRequest.Part filePart = new PostRequest.Part(MediaTypes.ZIP, report);
+    PostRequest post = new PostRequest("api/ce/submit")
+      .setMediaType(MediaTypes.PROTOBUF)
+      .setParam("projectKey", projectDefinition.getKey())
+      .setParam("projectName", projectDefinition.getName())
+      .setParam("projectBranch", projectDefinition.getBranch())
+      .setPart("report", filePart);
+    WsResponse response = wsClient.call(post).failIfNotSuccessful();
+    try (InputStream protobuf = response.contentStream()) {
+      return WsCe.SubmitResponse.parser().parseFrom(protobuf).getTaskId();
+    } catch (Exception e) {
```

Comment:
if I understand correctly, the `catch(Exception e)` is useless here. A try finally is enough